crates/dekaf/src/read.rs

```diff
@@ -267,12 +270,24 @@ impl Read {
 
             // Encode the value.
             let value =
-                if is_control || (is_deletion && matches!(self.deletes, DeletionMode::Default)) {
+                if is_control || (is_deletion && matches!(self.deletes, DeletionMode::Kafka)) {
                     None
                 } else {
                     tmp.push(0);
                     tmp.extend(self.value_schema_id.to_be_bytes());
-                    () = avro::encode(&mut tmp, &self.value_schema, root.get())?;
+
+                    if matches!(self.deletes, DeletionMode::CDC) {
+                        let mut heap_node = HeapNode::from_node(root.get(), &self.alloc);
+                        let foo = DELETION_INDICATOR_PTR
+                            .create_heap_node(&mut heap_node, &self.alloc)
```

Comment:
there's a memory leak here -- you're never resetting the bump allocator.

I would recommend _not_ having it as a field of `self` and instead creating one on the stack. Use it, and then call reset() on it immediately after each avro::encode so that it's only retaining a single HeapNode.

Reply:
I was going to ask you to take a look at this as I wasn't sure I was doing it correctly. That makes sense to me, thanks 👍